scripts/bootstrap-aio.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+#
+# Copyright 2014-2017, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+## Shell Opts ----------------------------------------------------------------
+set -e -u -x
+
+## Functions -----------------------------------------------------------------
+
+export BASE_DIR=${BASE_DIR:-"/opt/rpc-openstack"}
+source ${BASE_DIR}/scripts/functions.sh
+
+## Main ----------------------------------------------------------------------
+
+# Check the openstack-ansible submodule status
+check_submodule_status
+
+# Get minimum disk size
+DATA_DISK_MIN_SIZE="$((1024**3 * $(awk '/bootstrap_host_data_disk_min_size/{print $2}' ${OA_DIR}/tests/roles/bootstrap-host/defaults/main.yml) ))"
+# Determine the largest secondary disk device available for repartitioning which meets the minimum size requirements
+DATA_DISK_DEVICE=$(lsblk -brndo NAME,TYPE,RO,SIZE | \
+                   awk '/d[b-z]+ disk 0/{ if ($4>m && $4>='$DATA_DISK_MIN_SIZE'){m=$4; d=$1}}; END{print d}')
+# Only set the secondary disk device option if there is one
+if [ -n "${DATA_DISK_DEVICE}" ]; then
+  export BOOTSTRAP_OPTS="${BOOTSTRAP_OPTS} bootstrap_host_data_disk_device=${DATA_DISK_DEVICE}"
+fi
+
+# Run AIO bootstrap playbook
+if [ -z "${BOOTSTRAP_OPTS}" ]; then
```

Comment:
@hughsaunders Yeah this is just matching the bootstrap-aio.sh script in OSA. This check was introduced because of issues seen when executing ``-e ""`` which may have been temporary. I can clean this up in a follow on commit.